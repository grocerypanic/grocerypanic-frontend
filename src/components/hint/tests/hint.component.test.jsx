import React from "react";
import { render, cleanup } from "@testing-library/react";

import Hint from "../hint.component";

import Strings from "../../../configuration/strings";

describe("Setup Environment", () => {
  let utils;

  beforeEach(() => {
    utils = render(
      <Hint>{Strings.Testing.GenericMultiLineTranslationTestString}</Hint>
    );
  });

  afterEach(cleanup);

  it("should render with the correct message content", () => {
    const node = utils.getByTestId("HintDialogue");
    expect(node).toBeTruthy();
    expect(
      utils.getAllByText(Strings.Testing.GenericTranslationTestString).length
    ).toBe(2);
  });

  it("should match the snapshot on file (styles)", () => {
    expect(utils.container.firstChild).toMatchSnapshot();
  });
});
