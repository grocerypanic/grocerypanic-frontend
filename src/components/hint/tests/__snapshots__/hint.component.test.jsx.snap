// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Setup Environment should match the snapshot on file (styles) 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding-top: 5px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  font-size: 0.7em;
  color: #000000;
}

.c1 {
  text-align: center;
}

<div
  class="c0"
  data-testid="HintDialogue"
>
  <div
    class="c1"
  >
    GenericTranslationTestString
  </div>
  <div
    class="c1"
  >
    GenericTranslationTestString
  </div>
</div>
`;
